homeassistant/components/pylontech_us/config_flow.py

```diff
@@ -0,0 +1,119 @@
+"""Config flow for pylontech_us integration."""
+from __future__ import annotations
+
+import logging
+
+# from pickle import TRUE
+from typing import Any
+
+from pylontech import PylontechStack
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+# TOODOP adjust the data schema to the data that you need
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("port", default="/dev/ttyUSB0"): str,
+        vol.Required("baud", default=115200): int,
+        vol.Required("battery_count", default=7): int,
+    }
+)
+
+
+class PylontechHub:
+    """Communication to Pylontech Battery stack."""
+
+    def __init__(self, config) -> None:
+        """Initialize."""
+        self._config = config
+
+    def validate_config_input(self) -> None:
+        """Validate config options. Raise exception on error."""
+        # If you cannot connect:
+        # throw CannotConnect
+        # If the authentication is wrong:
+        # InvalidAuth
+        # config['port']
+
+        stack = PylontechStack(
+            device=self._config["port"],
+            baud=self._config["baud"],
+            manualBattcountLimit=self._config["battery_count"],
+        )
+        stack.update()
+
+        if stack is None:
```

Comment:
It `stack` is `None` then would the `update()` method would have failed above?

Reply:
Rewrite of pylontech lib to pass exception up to homeassistant in case of failing connection and update.